### ISSUE TITLE
Main menu `authorize` helper doesn't seem to do anything

### ISSUE THREAD

**Issue by iainbeeston** [avo-hq/avo#2440]
### Describe the bug

According to the [avo 3.x documentation](https://docs.avohq.io/3.0/menu-editor.html#using-authorization-rules), if you want to hide a resource from the main menu according to your authorization framework, you should use the `authorize` helper method inside a `visible` block. However, when I tried to do this the menu option is still visible, no matter what I do in my policy. For example, if I have in my pundit `UserPolicy`:

```ruby
  def index?
    false
  end
```

and I call:

```ruby
      resource :user, visible: -> { authorize(current_user, User, :index?, raise_exception: false) }
```

it still shows users in the main menu.

In fact I can pass any arguments to `authorize` and it doesn't raise an error, for example:

```ruby
      resource :user, visible: -> { authorize(1, 2, 3) }
```

When I looked into how this helper works, it uses the [authorisation service](https://github.com/avo-hq/avo/blob/main/lib/avo/services/authorization_service.rb), which doesn't look like it has actually been implemented? (It will always return `true` from `authorize` and `authorize_action` and `apply_policy` just returns the `query` passed in.)

### Steps to Reproduce
1. Given an app using avo, a custom main menu and any resource, load up avo and confirm that that resource appears in the main menu
2. Open the `avo.rb` initialiser and add a visible parameter to the resource that calls authorize, like this: `visible: -> { authorize(1, 2, 3, false, true) }`
2. Reload the page with avo in it
3. Confirm that the resource still appears in the menu
4. Add a policy for that resource and make it return false for the index method.
5. Open the `avo.rb` initialiser and change `visible` to authorize(current_user, <resource>, "index?", raise_exception: false)`
2. Reload the page with avo in it
3. Confirm that the resource still appears in the menu

### Expected behavior & Actual behavior

Avo should throw an error when it renders the main menu the first time. The second time it should hide the resource in the menu.

### System configuration
**Avo version**: 3.3.2

**Rails version**: 7.0.8

**Ruby version**: 3.2.2

**License type**:

 - [x] Community
 - [ ] Pro
 - [ ] Advanced

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

#### Impact

 - [x] High impact (It makes my app un-usable.)
 - [ ] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

There is a workaround for this, but others might be following the documentation and not realise that their authorisation is being ignored, which seems quite serious.

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [x] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [ ] Low urgency (It can wait. I just wanted you to know about it.)

**Comment by adrianthedev** [comment]
The authorization feature was always a [Pro tier](https://avohq.io/pricing) feature.

![CleanShot 2024-02-02 at 14 18 16@2x](https://github.com/avo-hq/avo/assets/1334409/b3179ee2-9a95-4b6b-a605-16bac3c60390)

**Comment by adrianthedev** [comment]
No problem @iainbeeston. 
Let me try to reproduce this on mu end

**Comment by Paul-Bob** [comment]
Reproduced and found the issue, thanks for the detailing report @iainbeeston, I'll open a PR with a fix soon.

**Comment by Paul-Bob** [comment]
@iainbeeston this should be fixed on `3.3.4` fresh release
